lib/xboxrt/libc_extensions/tchar.h

```diff
@@ -0,0 +1,101 @@
+#ifndef _XBOXRT_TCHAR_H
+#define _XBOXRT_TCHAR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(UNICODE) || defined(_MBCS)
+#error nxdk does not support the Unicode API
+#endif
+
+#define _TEOF EOF
+
+#ifndef __TCHAR_DEFINED
+#define __TCHAR_DEFINED
+typedef char _TCHAR;
+typedef int _TINT;
+typedef signed char _TSCHAR;
+typedef unsigned char _TUCHAR;
+typedef char _TXCHAR;
+#endif
+
+#define __T(x) x
+#define _T __T
+#define _TEXT __T
```

Comment:
nit: Does MS also use `__T` for this step? I'm a bit worried about side-effects (overwriting `__T` / defining a very generic name like `__T` that might conflict with some lib).

If MS doesn't use this name, then we could `#undef __T` afterwards; but it still might have side-effects (still ruining `__T`, so include order is relevant now).

Reply:
The MS header indeed uses `__T`. It's used to do the actual work (i.e. attaching the `L` prefix when `UNICODE` is defined: `#define __T(x) L ## x`), while `_T` and `_TEXT` then just reference it.